src/main.rs

```diff
@@ -106,15 +164,50 @@ async fn fetch_root_for_page(page: i32) -> Result<Root, reqwest::Error> {
 
 async fn process_paged_orders(root: &Root, page: i32) -> (i32, i32) {
     //process and handle exceptions per order
+    let (shopify_config, strapi_config) = load_configs();
     let mut processed: i32 = 0;
-    for _ in &root.data {
+    for data in &root.data {
         //sleep for 1s
         print!(".");
         tokio::time::sleep(std::time::Duration::from_secs(1)).await;
-        //replace this time with two calls to:
-        //1. Delete the order in Shopify is it exists
-        //2. Delete the order in Strapi
+        delete_shopify_resource(&shopify_config, "orders", data.id).await;
```

Comment:
Check if the reference exists in the data payload and then delete_shopify only after.